styx-common/src/main/java/com/spotify/styx/storage/BigtableStorage.java

```diff
@@ -169,6 +193,20 @@ WorkflowInstanceExecutionData executionData(WorkflowInstance workflowInstance) t
     return WorkflowInstanceExecutionData.fromEvents(events);
   }
 
+  private List<WorkflowInstanceExecutionData> executionData(
+      Set<WorkflowInstance> workflowInstancesSet) {
+    return workflowInstancesSet.parallelStream()
+        .map(workflowInstance -> {
+          try {
+            return executionData(workflowInstance);
+          } catch (IOException e) {
+            throw Throwables.propagate(e);
+          }
+        })
+        .sorted(WorkflowInstanceExecutionData.COMPARATOR)
```

Comment:
The reason it's a set is because we're iterating through all events from Bigtable, and each WorkflowInstance will have several events. So the set is needed.